torchvision/ops/poolers.py

```diff
@@ -153,14 +174,21 @@ def forward(self, x, boxes, image_shapes):
             device=device,
         )
 
+        results = []
         for level, (per_level_feature, scale) in enumerate(zip(x, self.scales)):
-            idx_in_level = torch.nonzero(levels == level).squeeze(1)
+            idx_in_level = torch.nonzero(levels.to(torch.int64) == level).squeeze(1)
```

Comment:
Can you explain why this cast is necessary? From https://github.com/pytorch/vision/blob/e8d3291f497c339b014a0b0c823621821c063823/torchvision/ops/poolers.py#L40
`levels` are already cast to `int64`, so this seems unecessary?

Reply:
self.k_min is float so the returned value was a torch.float. 
I added a cast on (target_lvls.to(torch.int64) - self.k_min) and reverted the one on levels.

Reply:
Oh I see, this is a regression introduced by type promotion, thanks a lot for the catch!

In this case, can you instead change https://github.com/pytorch/vision/blob/17e355f70267e0e01f7c4355a75d46b76f55b5aa/torchvision/ops/poolers.py#L111-L112
to do instead
```python
lvl_min = int(-torch.log2(torch.tensor(scales[0], dtype=torch.float32)))
lvl_max = int(-torch.log2(torch.tensor(scales[-1], dtype=torch.float32)))
```
i.e., replace `.item()` with a `int()`? This way it fixes it at the root cause of it